### PR TITLE
tests: add test to run snap inside lxd as a user

### PR DESCRIPTION
**Pull request by mvo5**
This reproduces the error reported in https://forum.snapcraft.io/t/snapcraft-adt-failures-with-the-new-core-release/2850/12

### CODE REVIEW

File: cmd/snap-confine/snap-confine.c [hunk 1]
```diff
@@ -131,8 +131,28 @@ int main(int argc, char **argv)
...
+	// change this is causing for the code temporairly set the effective group
+	// id to the real group id. We will change that again below when we
+	// manipulate the freezer cgroup.
+	// Because LXD uses particular permissions for cgroups we need to be
+	// group-root for that operation to succeed.
+	if (euid == 0 && ruid != 0) {
+		if (setegid(ruid) != 0) {
+			die("cannot set effective group id to %d", ruid);
+		}
+	}
```
[the remark this answers is not in the document]
Comment (zyga) [reply cmd/snap-confine/snap-confine.c]: Oh, man, thanks!

[the remark this answers is not in the document]
Comment (zyga) [reply cmd/snap-confine/snap-confine.c]: In retrospective the non abbreviated names are really really good.

File: cmd/snap-confine/snap-confine.c [hunk 2]
```diff
@@ -226,7 +240,21 @@ int main(int argc, char **argv)
...
+
+			if (getegid() != 0 && saved_gid == 0) {
+				// Temporarily raise egid so we can chown the freezer cgroup
+				// under LXD.
+				if (setegid(0) != 0) {
+					die("cannot set effective group id to root");
+				}
+			}
 			sc_cgroup_freezer_join(snap_name, getpid());
+			if (geteuid() == 0) {
```
Comment (jdstrand) [inline cmd/snap-confine/snap-confine.c]: Shoot, this should be (sorry this wasn't part of my original suggestion):
```
if (geteuid() == 0 && real_gid != 0) {
```

**Comment by mvo5** [comment]
Sorry, git went crazy on me, I try to clean this up, in the meantime: https://github.com/snapcore/snapd/pull/4246

**Pull request closed without merging**